### ISSUE TITLE
Does anyone use some kind of formula to set liquidation price to a specific price. 

### ISSUE THREAD

**Issue by Kalelv45** [ccxt/ccxt#6246]
Don't know if I have to post this question here, but I have the following. 

Let's say I have an Open position with a relative stop loss. 
Since in some cases slippage is very crazy. ( I had 1k slippage once on mex during high volatility). 
I would like to know if there is a formula to use where I can set my liquidation price a specific % under my stop_loss just to give it some breathing room. If it hit my liquidation price first the better. With isolated margin in some cases I think this is the best approach for me. 

Any contribution is highly appreciated.

**Comment by samgermain** [comment]
I've written the [python swap liquidation price formula for binance on Freqtrade](https://github.com/freqtrade/freqtrade/blob/9029833c8c7397424a0e97c1816634ca2289e2c0/freqtrade/exchange/binance.py#L128)

These are the URLs for the margin and swap liquidation price formulas for binance

MARGIN: https://www.binance.com/en/support/faq/f6b010588e55413aa58b7d63ee0125ed
SWAP: https://www.binance.com/en/support/faq/b3c689c1f50a44cabb3a84e663b81d93

I've also written the futures liquidation formula for [okex and gateio,](https://github.com/freqtrade/freqtrade/blob/9029833c8c7397424a0e97c1816634ca2289e2c0/freqtrade/exchange/exchange.py#L2496) but only the isolated liquidation price as of the time of writing this. These two turned out to be the same formula once manipulated, so my best guess is that other exchanges probably also use this liquidation formula. 

These are the URLs for gateio and okex's liquidation formulas

gateio: https://www.gate.io/help/futures/perpetual/22160/calculation-of-liquidation-price
okex: https://www.okex.com/support/hc/en-us/articles/
            360053909592-VI-Introduction-to-the-isolated-mode-of-Single-Multi-currency-Portfolio-margin
            
[Here I wrote something](https://github.com/freqtrade/freqtrade/blob/9029833c8c7397424a0e97c1816634ca2289e2c0/freqtrade/exchange/exchange.py#L2486) that does what you describe as give it some breathing room. I set a liquidation_buffer of some value like 0.05, I then set the minimum stop loss (maximum for shorts) using this formula

`stop_loss = liquidation_price ± (abs(open_rate - liquidation_price) * liquidation_buffer)`

    `± = + for long trades`
    `± = - for short trades`

I wrote a more in depth [explanation of this liquidation_buffer in the freqtrade documentation](https://www.freqtrade.io/en/latest/leverage/#understand-liquidation_buffer)

**Comment by samgermain** [comment]
I'm going to close this issue, feel free to reopen it if you haven't any questions
